src/keyboard/LayerPicker.tsx

```diff
@@ -30,10 +31,79 @@ interface LayerPickerProps {
   onLayerNameChanged?: (
     id: number,
     oldName: string,
-    newName: string
+    newName: string,
   ) => void | Promise<void>;
 }
 
+interface EditLabelData {
+  id: number;
+  name: string;
+}
+
+const EditLabelModal = ({
+  open,
+  onClose,
+  editLabelData,
+  handleSaveNewLabel,
+}: {
+  open: boolean;
+  onClose: () => void;
+  editLabelData: EditLabelData;
+  handleSaveNewLabel: (
+    id: number,
+    oldName: string,
+    newName: string | null,
+  ) => void;
+}) => {
+  const ref = useModalRef(open);
+  const labelInputRef = useRef<HTMLInputElement>(null);
+
+  const handleSave = () => {
+    const newName = labelInputRef.current?.value ?? null;
+    handleSaveNewLabel(editLabelData.id, editLabelData.name, newName);
+    onClose();
+  };
+
+  return (
+    <dialog
+      ref={ref}
+      defaultValue={editLabelData.name}
+      onClose={onClose}
+      className="p-5 rounded-lg border-text-base border min-w-min w-[30vw] flex flex-col"
+    >
+      <span className="mb-3 text-lg">Editing Layer Name</span>
+      <input
+        className="p-1 border rounded border-text-base border-solid"
+        ref={labelInputRef}
+        type="text"
+        defaultValue={editLabelData.name}
+        autoFocus
+        onKeyDown={(e) => {
+          if (e.key === "Enter") {
+            e.preventDefault();
+            handleSave();
+          }
+        }}
+      />
+      <div className="m-2 flex justify-end">
+        <button type="button" aria-label="Cancel Edit" onClick={onClose}>
+          <XMarkIcon className="h-7 w-7 mx-1 hover:text-accent" />
```

Comment:
We use text actions within the "Settings Reset" dialog, and this is using pure icons. I generally prefer text actions, so can we switch to that for consistency? I'm open to being convinced otherwise, but that feels better to me.

Reply:
sgtm!
if these buttons look good, i'll have a follow up PR to make primary and secondary button components so we don't always have to individually style buttons. We're using something similar to close the About/License modals.